### PR TITLE
Modularise menu

### PR DESCRIPTION
**Pull request by jleft**
A continuation of #157. Also adds:

- A top-level menu component, to handle the main menu
- Use of option generators moved out of main and into separate modules for each menu section
- A more generic 'option' class for use with the option generator

Still TODO: all of the logic for changing the data source is still in main - I think this should be pulled out into its own module - if we're in agreement, another issue can be created for this.

### CODE REVIEW

File: src/assets/js/menu/option/generator.js [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+                        .datum();
+                    dispatch.optionChange(selectedOption);
+                });
+        }
+
+        d3.rebind(optionGenerator, dispatch, 'on');
+
+        return optionGenerator;
+    };
+
```
Comment (djmiley) [inline src/assets/js/menu/option/generator.js]: In light of the `menu/option.js` file, and an option being a more generic structure, could this be renamed to 'menu/option/groupGenerator.js` or similar. For #187, I'm looking at the idea of a `toggleGenerator`, so this isn't the only generator.

File: src/assets/js/menu/option.js [hunk 2]
```diff
@@ -0,0 +1,11 @@
+(function(d3, fc) {
+    'use strict';
+    sc.menu.option = function(displayString, valueString, option) {
+        return {
+            displayString: displayString,
+            valueString: valueString,
+            option: option
+        };
+    };
```
Comment (djmiley) [inline src/assets/js/menu/option.js]: Was just thinking about this last night! :+1:

**Comment by djmiley** [comment]
I like this.

File: src/assets/js/menu/option/generator.js (hunk 1 above)
Comment (jleft) [inline src/assets/js/menu/option/generator.js]: Nice idea - how about `optionGroupGenerator`? Could then add `toggleGroupGenerator` later too?

Comment (djmiley) [inline src/assets/js/menu/option/generator.js]: SGTM

Comment (djmiley) [inline src/assets/js/menu/option/generator.js]: In the `menu` or `menu/option` directory?

Comment (jleft) [inline src/assets/js/menu/option/generator.js]: I was thinking `menu/option/group/generator.js` for the path and `sc.menu.option.group.generator` for the component name?

Comment (djmiley) [inline src/assets/js/menu/option/generator.js]: Yes, I like the sound of that. :+1:

Comment (jleft) [inline src/assets/js/menu/option/generator.js]: Done (882c997)

**Pull request merged**